Components/DayCalendar.js

```diff
@@ -9,50 +12,251 @@ var {
   ScrollView,
   View,
   Text,
+  PanResponder,
+  Image,
 } = React;
 
+type Event = {
+  title: string;
+  startAt: Date;
+  length: number;
+};
+
 var HOURS_COUNT = 25;
 var HOURS = [];
 for (var i = 0; i < HOURS_COUNT; i++) {
-    var hour = moment().hour(i).second(0).minute(0).format('h:mma');
+    var hour = moment().hour(i).minute(0).second(0).format('h:mma');
     HOURS.push(hour);
 }
+var MINUTES_INA_DAY = 60*24;
+
+var DayCalendar = React.createClass({
+
+  panResponder: {},
+  resizeResponder: {},
+  draggableViewPreviousTop: 0,
+  draggableViewPreviousHeight: 0,
+  draggableView: (null : ?{ setNativeProps(props: Object): void }),
+  resizeHandleView: (null : ?{ setNativeProps(props: Object): void }),
+  draggableViewTopAndHeight: {},
+  scrollView: (null : ?{ setNativeProps(props: Object): void }),
+  totalHeight: 0,
+  beginOfDay: null,
+  endOfDay: null,
+
+  propTypes: {
+    style: View.propTypes.style,
+    eventBoxStyle: View.propTypes.style,
+    day: React.PropTypes.instanceOf(Date).isRequired, 
+    events: React.PropTypes.array.isRequired,
+    newEvent: React.PropTypes.object,
+    timelineHeight: React.PropTypes.number.isRequired,
+    timelineGap: React.PropTypes.number.isRequired,
+    onNewEventStartAtChange: React.PropTypes.func,
+    onNewEventLengthChange: React.PropTypes.func,
+  },
+
+  getDefaultProps: function() {
+    return { day: new Date(), events: [], timelineHeight: 1, timelineGap: 59 };
+  },
+
+  getInitialState: function() {
+    return {
+      newEvent: this.props.newEvent
+    };
+  },
+
+  _onPropsChange: function(props) {
+    this.totalHeight = (HOURS_COUNT-1) * (props.timelineGap + props.timelineHeight);
+    this.beginOfDay = moment(props.day).hour(0).minute(0).second(0);
+    this.endOfDay = moment(props.day).hour(24).minute(0).second(0);
```

Comment:
For these two `moment` assignments, you could do:
```js
this.beginOfDay = moment(props.day).startOf('day');
this.endOfDay = moment(props.day).endOf('day').startOf('hour');
```

Also, `.hour(24)` would signify midnight of the following day -- is that what you meant? If so, i redact my above statement and suggestion instead:

```js
this.endOfDay = moment(props.day).add(1, 'day').startOf('hour');
```